### ISSUE TITLE
Decklink support

### ISSUE THREAD

**Issue by Sadoharu** [slyfox1186/ffmpeg-build-script#5]
is there a way to add decklink support to ffmpeg?

**Comment by slyfox1186** [comment]
Can you link me the ( git repo / source code URL) for this? Any other info will help me add it faster.

**Comment by Sadoharu** [comment]
The compilation files can be downloaded from here [Decklink drivers](https://www.blackmagicdesign.com/support/family/capture-and-playback)
![image_2023-10-14_08-51-49](https://github.com/slyfox1186/ffmpeg-build-script/assets/77967977/bef6a272-b6eb-40e7-b8c4-a035c7a81aa6)

I also already have downloads available at the link [Decklink SDK 12.7](https://fex.net/s/ssvyz7r)

there is also a short [compilation guide](https://gist.github.com/afriza/879fed4ede539a5a6501e0f046f71463)

**Comment by slyfox1186** [comment]
So compiling this from source produced a plethora of errors. It interacted with multiple libraries so I tried doing it on another OS using Ubuntu Jammy and it still threw similar errors albeit different. Not sure what my best path forward on this is at this time. I will continue to work on it however it looks bleak at the moment. Why don't you try to build it and report back to me on how you did.

**Comment by slyfox1186** [comment]
If you have success let me know otherwise I will close this.

**Comment by ma-blip** [comment]
unpack SDK zip and copy contents of Linux folder to ffmpeg-build-script/workspace/include/BMD_SDK

Add  --enable-decklink                           \  to the script. line 2888 or forward

add on line 154 at the end   -I${workspace}/include/BMD_SDK/include or if you want to link it from somewhere else makle sure to change the dir

CFLAGS="-I${workspace}/include -I/usr/local/include -I/usr/include -I/usr/include/x86_64-linux-gnu -I/usr/include/SDL2 -I${workspace}/include/BMD_SDK/include"

Compiles ok for me Ubuntu 22.04 lts. Dont have any card right now and have not installed any drivers so cant confirm more then the compile.

**Comment by slyfox1186** [comment]
Ok I will attempt to add this and if it works I will update the script. Thanks!

**Comment by ma-blip** [comment]
You should be able to automate the process by using these available Decklink headers:
https://gitlab.com/m-ab-s/decklink-headers
Regards

**Comment by slyfox1186** [comment]
I have created the code required to download and install the declink-headers that you posted earlier.

However, it throws an error saying it is missing the file `windows.h`.

I found a copy of that file and put it into the  `$workspace/include` folder which is where the decklink header files are also located and it still throws the same error.

Gonna keep trying as I have the time.

As ALWAYS if you have more advice it can only help speed things up.
